### ISSUE TITLE
3.0.0-rc1: GbXMLReverseTranslator produces incorrect geometry with respect to the windows and exterior walls of the gbxml

### ISSUE THREAD

**Issue by jmarrec** [NREL/OpenStudio#3951]
Issue overview
--------------

3.0.0-rc1: GbXMLReverseTranslator produces incorrect geometry with respect to the windows and exterior walls of the gbxml

On the other hand the 2.9.1 GbXMLReverseTranslator  produces completely normal geometry.

Reported by @antonszilasi 

## Current Behavior

It doesn't work with 3.0.0, but worked by 2.9.1

## Expected Behavior

It should produce valid geometry.

## Steps to Reproduce

The following XML exhibits the issue: 

[3951_Geometry_bug.xml.txt](https://github.com/NREL/OpenStudio/files/4501889/3951_Geometry_bug.xml.txt)

I can create an OSM for both 2.9.1 and 3.0.0-rc3 and compare the diffs:

```
C:\openstudio-2.9.1\bin\openstudio.exe -e "rt = OpenStudio::GbXML::GbXMLReverseTranslator.new; m = rt.loadModel('3951_Geometry_bug.xml').get; m.save('291.osm', true)"
```

```
C:\openstudio-3.0.0-rc3\bin\openstudio.exe -e "rt = OpenStudio::GbXML::GbXMLReverseTranslator.new; m = rt.loadModel('3951_Geometry_bug.xml').get; m.save('300.osm', true)"
```

**From a diff between the two, it appears that all `OS:Surface` in 3.0.0 have a 'Surface Type' of 'Wall' instead of RoofCeiling, Floors, etc.**

## Possible Solution

This is probably a regression introduced during the transition from QXml to pugixml.

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of OpenStudio (if using an intermediate build, include SHA): 3.0.0-rc1, 3.0.0-rc3

**Comment by jmarrec** [comment]
Found the issue in this commit : d436ec3c78e8559ddeb2e479d0928dd5a528f08f

There's a single block where the check to see if a string contains another is incorrect, and it's just a typo basically.

https://github.com/NREL/OpenStudio/blob/0cb319a4977599dbe3eca55fc39abdfa077aeaa7/src/gbxml/ReverseTranslator.cpp#L660

Couple of lines later it's done as it shouldn checking if the return value is not `npos`: https://github.com/NREL/OpenStudio/blob/0cb319a4977599dbe3eca55fc39abdfa077aeaa7/src/gbxml/ReverseTranslator.cpp#L688

**Comment by antonszilasi** [comment]
@jmarrec @tijcolem unfortunately this issue is not fixed I am still experiencing this issue

**Comment by jmarrec** [comment]
@antonszilasi the test file you had supplied works, the surface types are correctly assigned. Provide enough info to reproduce and I'll look into it.

**Comment by antonszilasi** [comment]
@jmarrec again thank you for your attention on this

To be more clear I am actually using openstudio:3.0.1-rc1 but I assumed that this issue ought to have been fixed in 3.0.0-rc1

None the less if you take the gbxml contained in the zip attached when you convert it to an osm you will see the following geometry
[SumitBansal-15_0.zip](https://github.com/NREL/OpenStudio/files/4795650/SumitBansal-15_0.zip)

![image](https://user-images.githubusercontent.com/8773483/84964306-03fc4180-b0c9-11ea-9add-6f751c99809f.png)

As you can see the windows are being scaled incorrectly.

I actually wrote to Luigi at NREL regarding this issue but I didnt realize that you were the coder behind squashing these bugs!

if you could fix this for openstudio:3.0.1-rc1 (or guide me how to) I'd be very grateful.

**Comment by jmarrec** [comment]
Gotcha. So the scaling should be fixed by #3998 (and I opened #3997 specifically for that issue with that picture and co).

**Comment by antonszilasi** [comment]
@jmarrec I guess we wait for @tijcolem to merge into develop then he can tag a new 3.0.1-rc1
